src/main/java/org/team1540/base/motionprofiling/MotionProfile.java

```diff
@@ -0,0 +1,173 @@
+package org.team1540.base.motionprofiling;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import jaci.pathfinder.Trajectory;
+import java.util.Set;
+
+public class MotionProfile extends Command {
+
+  private int slotId = 0;
+  private Set<Properties> motionProfiles;
+  private Timer timer = new Timer();
+  private double lastTime;
+  private boolean isFinished = false;
+
+  public MotionProfile(Set<Properties> motionProfiles) {
+    this.motionProfiles = motionProfiles;
+  }
+
+  public int getSlotId() {
+    return slotId;
+  }
+
+  public void setSlotId(int slotId) {
+    this.slotId = slotId;
+  }
+
+  public Set<Properties> getMotionProfiles() {
+    return motionProfiles;
+  }
+
+  public void setMotionProfiles(
+      Set<Properties> motionProfiles) {
+    this.motionProfiles = motionProfiles;
+  }
+
+  protected void initialize() {
+    timer.start();
+    lastTime = timer.get();
+    isFinished = false;
+  }
+
+  protected void execute() {
+    for (Properties currentProperty : motionProfiles) {
+      // Each controller's setpoint is calculated at a slightly different time, but this doesn't
+      // matter, since the motion profile is "continuous."
+      double velocity = getVelocitySetpoint(currentProperty, timer.get(), lastTime);
+      SmartDashboard.putNumber("calculatedVelocity", velocity);
+      currentProperty.getSetMotorVelocityFunction().accept(velocity);
+    }
+
+    lastTime = timer.get();
+  }
+
+  private double getVelocitySetpoint(Properties currentProperty, double currentTime,
+      double lastTime) {
+
+    /*
+      Whoa! This is weird. Although everything is ordered base on time, that's prone to getting off
+      and just never correcting.
+      However, positions can collide. Thus, we search forward in time from the last point we
+      calculated until we get a position we like.
+
+      Note that there's no guessing what the velocity setpoint should be based on how long it took
+      the last loop to complete–that is, if your jerk is high, you might have some issues with the
+      setpoint being a little wonky, as though it's lagging. Otherwise you shouldn't really notice.
+    */
+
+    Trajectory thisTrajectory = currentProperty.getTrajectory();
+    double dt = thisTrajectory.segments[0].dt;
+    double encoderMultiplier = currentProperty.getEncoderTickRatio();
+    double quadraturePosition = currentProperty.getGetEncoderPositionFunction().getAsDouble();
+
+    // Start from the current time and find the closest point.
+    int startIndex = Math.toIntExact(Math.round(currentTime / dt));
+
+    SmartDashboard.putNumber("startIndex", startIndex);
+
+    // Moved away from this because of the additional complexity and unsolved problems
```

Comment:
Same with massive blocks of commented code.

Reply:
Keeping that for now since we (ideally) plan on implementing that properly in the future, same as the weird unused bits being passed around.

Reply:
Um no, if you want you can cherry-pick it off into a separate dev branch but not in library code.